File: VIQA/Common/Columns.cs
using System;
using System.Collections.Generic;
using System.Linq;
using OpenQA.Selenium;
using VIQA.HtmlElements;
using VIQA.HtmlElements.Interfaces;
using VIQA.HtmlElements.SimpleElements;
using VIQA.SiteClasses;

namespace VIQA.Common
{
    public class Columns<T> : TableLine<T> where T : VIElementsSet, IHaveValue
    {
        public Columns()
        {
            GetHeadersFunc = t => t.FindElements(By.XPath(".//th")).Select(el => el.Text).ToArray();
            HaveHeaders = true;
            ElementIndex = ElementIndexType.Nums;
        }
        private Exception GetRowsException(string colName, Exception ex)
        {
            return VISite.Alerting.ThrowError(string.Format("Can't Get Column '{0}'. Exception: {1}", colName, ex));
        }

        public override List<Cell<T>> this[string name]
        {
            get
            {
                try { return Table.Rows.Headers.Select(rowName => Table.Cell(name, rowName)).ToList(); }
                catch (Exception ex) { throw GetRowsException(name, ex); } 
            }
        }
        
        public override List<Cell<T>> this[int num]
        {
            get
            {
                int? colsCount = null;
                if (_count != null)
                    colsCount = _count;
                else if (_headers != null && _headers.Any())
                    colsCount = _headers.Count();
                if (colsCount != null && colsCount < num)
                    throw VISite.Alerting.ThrowError(string.Format("Can't Get Column '{0}'. [num] > ColumnsCount({1}).", num, colsCount));
                try
                {
                    var result = new List<Cell<T>>();
                    for (int rowNum = 1; rowNum <= Table.Rows.Count; rowNum++)
                        result.Add(Table.Cell(num, rowNum));
                    return result.ToList();
                }
                catch (Exception ex) { throw GetRowsException(num.ToString(), ex); } 
            }
        }
    }
}

File: Java/src/main/java/ru/viqa/ui_testing/elements/baseClasses/Table/Columns.java
package ru.viqa.ui_testing.elements.baseClasses.Table;

import ru.viqa.ui_testing.page_objects.VISite;
import ru.viqa.ui_testing.elements.interfaces.IHaveValue;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;

import java.util.ArrayList;
import java.util.List;

import static ru.viqa.ui_testing.common.utils.LinqUtils.select;
import static java.lang.String.format;

/**
 * Created by 12345 on 26.10.2014.
 */
public class Columns<T extends IHaveValue> extends TableLine<T> {
    public Columns() {
        HaveHeaders = true;
        ElementIndex = ElementIndexType.Nums;
    }

    protected String[] getHeadersAction() throws Exception {
        return select(Table.getWebElement().findElements(By.xpath(".//th")), WebElement::getText)
                .toArray(new String[1]);
    }

    private Exception getRowsException(String colName, Exception ex) throws Exception {
        return VISite.Alerting.throwError(format("Can't Get Column '%s'. Exception: %s", colName, ex));
    }

    public final List<Cell<T>> getRow(String name) throws Exception {
        try { return new ArrayList<>(select(Table.getRows().getHeaders(), rowName -> Table.cell(name, rowName))); }
        catch (Exception ex) { throw getRowsException(name, ex); }
    }

    public List<Cell<T>> getRow(int num) throws Exception {
        int colsCount = -1;
        if (_count > 0)
        colsCount = _count;
        else if (_headers != null && (_headers.length > 0))
        colsCount = _headers.length;
        if (colsCount > 0 && colsCount < num)
        throw VISite.Alerting.throwError(format("Can't Get Column '%s'. [num] > ColumnsCount(%s).", num, colsCount));
        try {
            List<Cell<T>> result = new ArrayList<>();
            for (int rowNum = 1; rowNum <= Table.getRows().getCount(); rowNum++)
                result.add(Table.cell(num, rowNum));
            return result;
        }
        catch (Exception ex) { throw getRowsException(num + "", ex); }
    }
}
